src/Client.php

```diff
@@ -85,14 +93,18 @@ public function getSettings(): Settings
     /**
      * {@inheritdoc}
      */
-    public function devPing(?string $value = null): DevPingResponse
+    public function devPing(?DevPingParams $params = null): DevPingResponse
     {
+        $query = [
+            'value' => (string) \time(),
+        ];
+
+        if ($params instanceof DevPingParams && is_string($value = $params->getValue())) {
+            $query['value'] = $value;
+        }
+
         return DevPingResponse::fromHttpResponse(
-            $this->doRequest(new Request('get', 'dev/ping'), [
-                'query' => [
-                    'value' => $value ?? ((string) \time()),
-                ],
-            ])
+            $this->doRequest(new Request('get', 'dev/ping'), ['query' => $query])
```

Comment:
@tarampampam 

Now inside `$this->doRequest(...)` we do:
- Build request URL
- Make a request options (often very large, with logic)

This is **more verbose**, but I suggest following the scheme:
- Build query URL, if with logic
- Set default query options
- Modify options, if needed
- Send request